python/sdk/merlin/model.py

```diff
@@ -1115,6 +1105,62 @@ def list_endpoint(self) -> List[VersionEndpoint]:
             endpoints.append(VersionEndpoint(ep))
         return endpoints
 
+    def build_image(self, resource_request: Optional[ResourceRequest] = None):
+        """
+        Build the Docker image for this model version.
+
+        :param resource_request: The resource requirement (CPU & memory) for image building job.
+
+        :return: VersionImage object
+        """
+        target_resource_request = None
+        if resource_request is not None:
+            target_resource_request = client.ResourceRequest(
+                cpu_request=resource_request.cpu_request,
+                memory_request=resource_request.memory_request,
+            )
+
+        options = client.BuildImageOptions(
+            resource_request=target_resource_request,
+        )
+
+        version_image_api = VersionImageApi(self._api_client)
+        version_image_api.models_model_id_versions_version_id_image_put(
+            model_id=self.model.id,
+            version_id=self.id,
+            body=options,
+        )
+
+        bar = pyprind.ProgBar(
+            100,
+            track_time=True,
+            title=f"Building Docker image for model {self.model.name} version {self.id}",
+        )
+
+        while True:
```

Comment:
Wait, if the image building job fails or gets interrupted somehow, would this loop never end (until 10s * 100 iterations or around 16/17 minutes later)? :O

Reply:
Yes, because the current implementation is not storing any state on the DB, we don't have any reliable source for image existence other than the image repository itself. We depends on the remote repository for the image status.

If we want to have a reliable source about the image building status, I think we should have a new table for this (I don't think we can reuse the deployments table as it has endpoint_status which is specifically used for the version endpoint).

Reply:
As discussed offline, I've updated the Get Image API to get image building k8s job as well so that this poll will check the `image_building_job_status` whether to continue polling or break.